File: problemsets/Codeforces/C++/B946.cpp
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
#include <bits/stdc++.h>

using namespace std;

#define eb emplace_back
#define fi first
#define se second
typedef pair<int,int> pii;

#define sqr(x) (x)*(x)

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);
	long long a, b; cin >> a >> b;
	while (a && b && (a >= b<<1 || b >= a<<1))
	  tie(a,b) = make_pair(a%(b<<1), b%(a<<1));
	cout << a << " " << b;
}

File: problemsets/Codeforces/Python/B946.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
a,b=map(int,input().split())
while 1:
  if a==0 or b==0:break
  if a>=2*b:
    a=a%(2*b)
    continue
  if b>=2*a:
    b=b%(2*a)
    continue
  break
print(a,b)